c/tskit/trees.c

```diff
@@ -2012,6 +2018,113 @@ tsk_treeseq_trait_correlation(tsk_treeseq_t *self,
     return ret;
 }
 
+static int
+trait_regression_summary_func(size_t state_dim, double *state, size_t result_dim,
+        double *result, void *params)
+{
+    covariates_stat_params_t args = *(covariates_stat_params_t *) params;
+    const double num_samples = (double) args.num_samples;
+    const tsk_size_t k = args.num_covariates;
+    const double *V = args.V;;
+    const double *x = state;
+    const double *v;
+    double m, a, denom, xx;
+    size_t i, j;
+
+    m = x[state_dim - 1];
+    for (i = 0; i < result_dim; i++) {
+        if ((m > 0.0) && (m < num_samples)) {
+            v = GET_2D_ROW(V, k, i);
+            a = x[i];
+            denom = m;
+            for (j = 0; j < k; j++) {
+                xx = x[result_dim + j];
+                a -= xx * v[j];
+                denom -= xx * xx;
+            }
+            if (denom < 1e-8) {
+                result[i] = 0.0;
+            } else {
+                result[i] = (a * a) / (denom * denom) / 2;
```

Comment:
Is this the same as ``(a * a) / (2 * denom * denom)``? Or is it ``(2 * a * a) / (denom * denom)``. It'd be good for readability to either put in some brackets or reorganise it a bit.

Reply:
whoops, yes!